app/navigation/CategorisationNavigator.scala

```diff
@@ -222,9 +237,23 @@ class CategorisationNavigator @Inject() (categorisationService: CategorisationSe
       case None => controllers.problem.routes.JourneyRecoveryController.onPageLoad()
     }
 
-  private def shouldGoToLongerCommodityCodeFromPrepPage(catInfo: CategorisationInfo, scenario: Scenario) =
-    catInfo.getMinimalCommodityCode.length == minimumLengthOfCommodityCode && catInfo.descendantCount != 0 &&
-      getResultAsInt(scenario) == Category2AsInt
+  private def shouldGoToLongerCommodityCodeFromPrepPage(
+    catInfo: CategorisationInfo,
+    scenario: Scenario,
+    answers: UserAnswers,
+    recordId: String
+  ): Boolean =
+    answers.get(HasLongComCodeQuery(recordId)) match {
+      case Some(hasLongComCode) =>
+        !hasLongComCode &&
+          catInfo.getMinimalCommodityCode.length == minimumLengthOfCommodityCode &&
+          catInfo.descendantCount != 0 &&
+          getResultAsInt(scenario) == Category2AsInt
+      case None                 =>
+        throw new RuntimeException(
+          s"Unsure as to whether the com code is long (${catInfo.commodityCode}) so cannot decided where to go from prep page"
+        )
```

Comment:
I'd recommend handling this by having a place in the journey that we send them if it isn't set, if possible.

So for example if it isn't set can we redirect the user to CategorisationPreparationController where it gets set rather than letting it fail here?

Reply:
The standard thing to do I guess would be to send the to the journey recovery page which would redirect them to the start of the categorisation journey again.